gapis/server/grpc.go

```diff
@@ -87,24 +87,25 @@ func (s *grpcServer) stopIfIdle(ctx context.Context, server *grpc.Server, keepAl
 	// no activity from the client in a contiguous N chunks of time.
 	// This avoids killing the server if the machine is suspended (where the
 	// client cannot send hearbeats, and the system clock jumps forward).
-	const maxMissCount = 3
-	idleTimeout /= maxMissCount
-	misses := 0
+	const waitTime = time.Second
+	var idleTime time.Duration = 0
 
 	defer server.GracefulStop()
 	for {
 		select {
 		case <-task.ShouldStop(ctx):
 			return
-		case <-time.After(idleTimeout):
-			misses++
-			if misses >= maxMissCount {
-				log.W(ctx, fmt.Sprintf("Stopping GAPIS server as it has been idle for more than %v (--idle-timeout)", idleTimeout))
-				time.Sleep(time.Second * 3) // Wait a little in the hope this message makes its way to the client(s).
+		case <-time.After(waitTime):
+			idleTime += waitTime
+			if idleTime >= idleTimeout {
+				log.E(ctx, fmt.Sprintf("Stopping GAPIS server as it has not received heartbeat (--idle-timeout %v)", idleTimeout))
```

Comment:
It seems clear to you because you are looking at this code. It's less clear to someone who's using the public interface. I don't see how it makes any suggestion of restarting?